src/pages/secrets/SecretDetails.jsx

```diff
@@ -80,6 +84,25 @@ const SecretDetails = () => {
     setSecretKeyClicked("");
   };
 
+  const handleUpdateFileLoc = (path) => {
+    return new Promise((resolve, reject) => {
+      dispatch(increment("pendingRequests"));
+      client.secrets.setFileLocation(projectID, secretName, { rootPath : path}).then(() => {
+        const newPath = secrets.map(obj => {
+          if (obj.name !==  secretName) return obj;
+          const newData = Object.assign({}, { rootPath: path });
+          return Object.assign({}, secret, newData);
```

Comment:
Instead of using secret here which is defined outside this function you can use the `obj` variable which is scoped within this function itself. That way even if we rename the above secret variable we wouldn't have to change this function